File: src/test/java/com/github/kayjamlang/tests/expressions/IsExpressionTest.java
package com.github.kayjamlang.tests.expressions;

import com.github.kayjamlang.core.expressions.Expression;
import com.github.kayjamlang.core.KayJamLexer;
import com.github.kayjamlang.core.KayJamParser;
import com.github.kayjamlang.core.Type;
import com.github.kayjamlang.core.expressions.ValueExpression;
import com.github.kayjamlang.core.expressions.IsExpression;
import com.github.kayjamlang.tests.TestsUtils;
import org.junit.BeforeClass;
import org.junit.Test;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertNotNull;

public class IsExpressionTest {

    @Test
    public void test() throws Exception {
        Expression expression = TestsUtils.parse("123 is int");
        assertNotNull(expression);
        assertEquals(IsExpression.class, expression.getClass());

        IsExpression isExpression = (IsExpression) expression;
        assertEquals(ValueExpression.class, isExpression.expression.getClass());
        assertEquals(Type.INTEGER, isExpression.verifyType);
    }
}

File: src/test/scala/com/github/kayjamlang/tests/expressions/IsExpressionTest.scala
package com.github.kayjamlang.tests.expressions

import com.github.kayjamlang.core.{KayJamLexer, KayJamParser, Type}
import com.github.kayjamlang.core.expressions.{IsExpression, ValueExpression}
import org.junit.Assert.{assertEquals, assertNotNull}
import org.junit.{BeforeClass, Test}

object IsExpressionTest {
  private var parser: KayJamParser = null

  @BeforeClass def prepare(): Unit = {
    parser = new KayJamParser(new KayJamLexer("123 is int"))
  }
}

class IsExpressionTest {
  @Test
  @throws[Exception]
  def test(): Unit = {
    val expression = IsExpressionTest.parser.readExpression
    assertNotNull(expression)
    assertEquals(classOf[IsExpression], expression getClass)
    val isExpression = expression.asInstanceOf[IsExpression]
    assertEquals(classOf[ValueExpression], isExpression.expression getClass)
    assertEquals(Type.INTEGER, isExpression.verifyType)
  }
}
